src/integration-test/integration/ODataClient.test.ts

```diff
@@ -0,0 +1,109 @@
+import "reflect-metadata";
+import { accessToken, projectId, sleep, testIModel, testIModelGroup } from "../utils";
+use(chaiAsPromised);
+
+describe("OData Client", () => {
+  const oDataClient: ODataClient = new ODataClient();
+  const reportsClient: ReportsClient = new ReportsClient();
+  const mappingsClient: MappingsClient = new MappingsClient();
+  const extractionClient: ExtractionClient = new ExtractionClient();
+
+  let reportId: string;
+  let oDataItem: ODataItem;
+  let mappingId: string;
+
+  before(async () => {
+    const newMapping: MappingCreate = {
+      mappingName: "Test",
+    };
+    const mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    mappingId = mapping.id;
+
+    const newGroup: GroupCreate = {
+      groupName: "Test",
+      query: "select * from biscore.element limit 10",
+    };
+    await mappingsClient.createGroup(accessToken, testIModel.id, mapping.id, newGroup);
+
+    const newReport: ReportCreate = {
+      displayName: "Test",
+      projectId,
+    };
+    const report = await reportsClient.createReport(accessToken, newReport);
+    reportId = report.id;
+
+    const newReportMapping: ReportMappingCreate = {
+      mappingId: mapping.id,
+      imodelId: testIModel.id,
+    };
+    await reportsClient.createReportMapping(accessToken, report.id, newReportMapping);
+
+    const extraction = await extractionClient.runExtraction(accessToken, testIModel.id);
+    let state = ExtractorState.Queued;
+    let status: ExtractionStatus;
+    for (const start = performance.now(); performance.now() - start < 6 * 60 * 1000; await sleep(3000)) {
+      status = await extractionClient.getExtractionStatus(accessToken, extraction.id);
+      state = status.state;
+      if(state !== ExtractorState.Queued && state.valueOf() !== ExtractorState.Running)
+        break;
+    }
+    expect(state).to.be.eq(ExtractorState.Succeeded);
+
+    const oDataResponse = await oDataClient.getODataReport(accessToken, reportId);
+    oDataItem = oDataResponse.value[0];
+  });
+
+  after(async () => {
+    await mappingsClient.deleteMapping(accessToken, testIModel.id, mappingId);
+    await reportsClient.deleteReport(accessToken, reportId);
+
+    await testIModelGroup.cleanupIModels();
+  });
+
+  it("get OData report", async () => {
+    const oDataResponse = await oDataClient.getODataReport(accessToken, reportId);
+    expect(oDataResponse).to.not.be.undefined;
+    expect(oDataResponse["@odata.context"]).to.not.be.empty;
+  });
+
+  it("get OData report metadata", async () => {
+    const oDataResponse = await oDataClient.getODataReportMetadata(accessToken, reportId);
+    expect(oDataResponse).to.not.be.undefined;
+    expect(oDataResponse).to.not.be.empty;
+    expect(oDataResponse[0].name).to.be.eq("Test_Test_Mapping_8d13eeb5d47d489a93a7f2f2a04eb075");
```

Comment:
This assertion should fail, because it contains Mapping Id and a new mapping is created for every test run. Change this assertion to check if the table name starts with `Test_Test`.

Reply:
changed assertion